src/qibolab/_core/parameters.py

```diff
@@ -202,3 +223,121 @@ def replace(self, update: Update) -> "Parameters":
+
+
+def _gate_sequence(qubit: Qubit, gate: str) -> Native:
+    """Default sequence corresponding to a native gate."""
+    channel = _gate_channel(qubit, gate)
+    pulse = Pulse(duration=0, amplitude=0, envelope=Rectangular())
+    if gate != "MZ":
+        return Native([(channel, pulse)])
+
+    return Native(
+        [(channel, Readout(acquisition=Acquisition(duration=0), probe=pulse))]
+    )
+
+
+def _pair_to_qubit(pair: str, qubits: QubitMap) -> Qubit:
+    """Get first qubit of a pair given in ``{q0}-{q1}`` format."""
+    q = tuple(pair.split("-"))[0]
+    try:
+        return qubits[q]
+    except KeyError:
+        return qubits[int(q)]
+
+
+def _native_builder(cls, qubit: Qubit, natives: set[str]) -> NativeContainer:
+    """Build default native gates for a given qubit or pair.
+
+    In case of pair, ``qubit`` is assumed to be the first qubit of the pair,
+    and a default pulse is added on that qubit, because at this stage we don't
+    know which qubit is the high frequency one.
+    """
+    return cls(
+        **{
+            gate: _gate_sequence(qubit, gate)
+            for gate in cls.model_fields.keys() & natives
+        }
+    )
+
+
+def _channel_config(id: ChannelId, channel: Channel) -> dict[ChannelId, Config]:
+    """Default configs correspondign to a channel."""
+    if isinstance(channel, DcChannel):
+        return {id: DcConfig(offset=0)}
+    if isinstance(channel, AcquisitionChannel):
+        return {id: AcquisitionConfig(delay=0, smearing=0)}
+    if isinstance(channel, IqChannel):
+        configs = {id: IqConfig(frequency=0)}
+        if channel.lo is not None:
+            configs[channel.lo] = OscillatorConfig(frequency=0, power=0)
+        if channel.mixer is not None:
+            configs[channel.mixer] = IqMixerConfig(frequency=0, power=0)
+        return configs
+    return {id: Config()}
+
+
+def initialize_parameters(
+    hardware: Hardware,
+    natives: Optional[set[str]] = None,
+    pairs: Optional[list[str]] = None,
+) -> Parameters:
+    """Generates default ``Parameters`` for a given hardware configuration."""
+    if natives is None:
+        natives = set()
+    else:
+        natives = set(natives)
+
+    configs = {}
+    for instrument in hardware.instruments.values():
+        if hasattr(instrument, "channels"):
+            for id, channel in instrument.channels.items():
+                configs |= _channel_config(id, channel)
+
+    single_qubit = {
+        q: _native_builder(SingleQubitNatives, qubit, natives - {"CP"})
+        for q, qubit in hardware.qubits.items()
+    }
+    coupler = {
+        q: _native_builder(SingleQubitNatives, qubit, natives & {"CP"})
+        for q, qubit in hardware.couplers.items()
+    }
+    if pairs is not None:
+        two_qubit = {
+            pair: _native_builder(
+                TwoQubitNatives, _pair_to_qubit(pair, hardware.qubits), natives
+            )
+            for pair in pairs
+        }
+    else:
+        two_qubit = {}
```

Comment:
Even this may be a non-conditional assignment with a conditioned value :)

Reply:
fcbfddd0fa69d50bb357c3fc7535c3ea18ce2292